### PR TITLE
Migrate Edge calls for Worker to FastAPI part 4 - Cleanup

### PR DESCRIPTION
**Pull request by jscheffl**
After PR https://github.com/apache/airflow/pull/44311, https://github.com/apache/airflow/pull/44330 and #44433 this PR is a cleanup and removes the old Internal API extensions.

For review: Only the last commit is relevant, the other 7 are from referenced base PRs

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: providers/src/airflow/providers/edge/worker_api/routes/_v2_routes.py [hunk 1]
```diff
@@ -20,12 +20,15 @@
...
-from functools import cache
-from typing import TYPE_CHECKING, Any, Callable
+from typing import TYPE_CHECKING, Any
 from uuid import uuid4
 
 from flask import Response, request
 
+from airflow.api_internal.endpoints.rpc_api_endpoint import (  # type: ignore[attr-defined]
+    # Note: This is just for compatibility with Airflow 2.10, not working for Airflow 3 / main as removed
+    initialize_method_map,
```
Comment (jscheffl) [inline providers/src/airflow/providers/edge/worker_api/routes/_v2_routes.py]: @potiuk FYI in my cleanup I realized... as the internal API is "gone" (virtually, at least not be usable, will contribute in a moment to clean as well...) and as workaround a copy was made... actually that part of code is only working in Airflow 2.10. Therefore removing the copy of the full method map here in the cleanup and sourcing this from the core (=2.10)

This way the "traces" are reduced, just a import from the past is used until we drop 2.10 support in Edge.

File: providers/src/airflow/providers/edge/CHANGELOG.rst [hunk 2]
```diff
@@ -27,6 +27,22 @@
 Changelog
 ---------
 
+0.9.0pre0
```
Comment (potiuk) [inline providers/src/airflow/providers/edge/CHANGELOG.rst:30]: Aren't we supposed to merge these two ?

**Review by potiuk — APPROVED** [review]
There is some devilish trickery with this one... Seems like it's no change at all:

Screenshot 2024-11-30 at 23 43 14

But ... It looks good:

**Comment by jscheffl** [comment]
> There is some devilish trickery with this one... Seems like it's no change at all:
> But ... It looks good:

Now after Part 3 is merged looks much better:

image

File: providers/src/airflow/providers/edge/CHANGELOG.rst (hunk 2 above)
Comment (jscheffl) [reply providers/src/airflow/providers/edge/CHANGELOG.rst:30]: It is an "internal breaking change" to remove these bindings. So I thought it is valid to get to a 0.9.0... but don't fear this means we are close to 1.0... still I assume there will be a 0.10.0 and 0.11.0 probably :-D

**Comment by potiuk** [comment]
> Now after Part 3 is merged looks much better:


:)

**Pull request merged**